### DISCUSSION TITLE
How to implement finite input set MPC? // mixed integer optimization

### DISCUSSION THREAD

**Discussion by tsbyq** [do-mpc/do-mpc#332] [Q&A] [answered]
Hi, 

Thank you for the great package and very helpful documentations. 

I was able to implement a heat pump controller following the tutorial. Now I am trying to implement it more realistically with on/off actions. I am curious if this is possible with `do-mpc` or anyone has experience with it.

Thanks,
Han

**Comment by 4flixt** [comment, marked as answer]
Dear Han, 
Thank you for the compliment 😀 
Unfortunately, do-mpc does currently not support mixed integer optimization which would be required for discrete (on/off) states. 

Best,
4flixt
